### ISSUE TITLE
"."(current directory)를 PATH 환경변수에 넣으면 안되는 이유

### ISSUE THREAD

**Issue by eubnara** [eubnara/study#42]
## tl;dr(요약)
 
  - PATH 환경변수 앞에 존재해서는 안된다.
    - 흔히 쓰이는 `ls` 와 같은 이름의 악의적인 프로그램을 만들어 두고, superuser 가 그걸 실행시키도록 유도할 수 있음
  -  PATH 환경변수의 마지막에 위치해서도 안된다.
    - `ls`, `more` 과 같은 명령어를 현재 디렉터리에 있는 걸 우선적으로 실행시키진 않겠지만, `mroe` 와 같은 프로그램을 파일시스템 전역에 뿌려놓았을 때, superuser 가 오타로 `mroe` 를 입력하였을 때, 그 프로그램이 실행될 수 있다.

## 본문

(Practical UNIX and Internet Security, 3rd Edition 에서 발췌)
STEALING SUPERUSER
Once upon a time, many years ago, one of us needed access to the root account on an academic machine. Although we had been authorized by management to have root access, the local system manager didn’t want to disclose the password. He asserted that access to the root account was dangerous (correct), that he had far more knowledge of Unix than we did (unlikely), and that we didn’t need the access (incorrect). After several diplomatic and bureaucratic attempts to get access normally, we took a slightly different approach, with management’s wry approval.

We noticed that this user had “.” at the beginning of his shell search path. This meant that every time he typed a command name, the shell would first search the current directory for the command of the same name. When he did a su to root, this search path was inherited by the new shell. This was all we really needed.

First, we created an executable shell file named ls in the current directory:

```
#!/bin/sh
cp /bin/sh ./stuff/junk/.superdude
chmod 4555 ./stuff/junk/.superdude
rm -f $0
exec /bin/ls ${1+"$@"}
Then, we executed the following commands:
```
```
% cd 
% chmod 700 .
% touch ./-f
```

The trap was ready. We approached the recalcitrant administrator with the complaint, “I have a funny file in my directory I can’t seem to delete.” Because the directory was mode 700, he couldn’t list the directory to see the contents. So, he used su to become user root. Then he changed the directory to our home directory and issued the command ls to view the problem file. Instead of the system version of ls, he ran our version. This created a hidden setuid root copy of the shell, deleted the bogus ls command, and ran the real ls command. The administrator never knew what happened.

We listened politely as he explained (superciliously) that files beginning with a dash character (-) needed to be deleted with a pathname relative to the current directory (in our case, rm ./-f); of course, we knew that.

A few minutes later, he couldn’t get the new root password.

## 참고
 
  - Practical UNIX and Internet Security, 3rd Edition
    - Chapter 5, STEALING SUPERUSER
    - Chapter 16
    - Chapter 23
